### PR TITLE
checking if the _firstOperand === to an operator & removing the codes…

### PR DESCRIPTION
**Pull request by yaaHosein**
Let us work to work together and develop the web 😢😢💔

### CODE REVIEW

File: calculator.js [hunk 1]
```diff
@@ -97,30 +99,25 @@ periodButton.addEventListener("click", function() {
 negativeValueButton.addEventListener("click", function() {
     const currentValueOfResult = result.textContent;
     if (currentValueOfResult !== "" && currentValueOfResult[0] !== "-") { result.textContent = "-" + currentValueOfResult };
-    if (currentValueOfResult !== "" && currentValueOfResult[0] === "-") { result.textContent = currentValueOfResult.slice(1) };
-
+    if (currentValueOfResult !== "" && currentValueOfResult[0] === "-") { result.textContent = currentValueOfResult.slice(0) };
```
Comment (SMH110) [inline calculator.js]: Why the slice value changes to 0?

File: calculator.js [hunk 2]
```diff
@@ -14,78 +15,79 @@ button0.addEventListener("click", function() {
...
+    if (_firstOperand === "+" || "-" || "*" || "/") result.textContent = "";
 });
 
 button1.addEventListener("click", function() {
     const currentValueOfResult = result.textContent;
     const newValue = currentValueOfResult + 1;
     console.log(parseInt(newValue));
     result.textContent = newValue;
-
+    if (_firstOperand === "+" || "-" || "*" || "/") result.textContent = "";
```
Comment (SMH110) [inline calculator.js]: This check need to be done at the beginning of the function

File: calculator.js [hunk 3]
```diff
@@ -97,30 +99,25 @@ periodButton.addEventListener("click", function() {
 negativeValueButton.addEventListener("click", function() {
     const currentValueOfResult = result.textContent;
     if (currentValueOfResult !== "" && currentValueOfResult[0] !== "-") { result.textContent = "-" + currentValueOfResult };
-    if (currentValueOfResult !== "" && currentValueOfResult[0] === "-") { result.textContent = currentValueOfResult.slice(1) };
-
+    if (currentValueOfResult !== "" && currentValueOfResult[0] === "-") { result.textContent = currentValueOfResult.slice(0) };
 });
 
 addingButton.addEventListener("click", function() {
```
Comment (SMH110) [inline calculator.js:117]: You're not using `_firstOperand ` in the operator functions 
you need to update the value of `_firstOperand ` with the operator type

File: calculator.js (hunk 1 above)
Comment (yaaHosein) [reply calculator.js]: It's not its first time 
maybe there was pulling problems !!

File: calculator.js (hunk 2 above)
Comment (yaaHosein) [reply calculator.js]: ok thanks

File: calculator.js (hunk 3 above)
Comment (yaaHosein) [reply calculator.js:117]: yes i am 
i will think how

File: calculator.js (hunk 1 above)
Comment (SMH110) [reply calculator.js]: Can you check what the value of this on your local master branch ?

Comment (yaaHosein) [reply calculator.js]: I knew why It became 0 and the reason behind it was that when i was facing branch and git hub errors I copied all the codes in a new notepad file 
then when i fixed the branch problems 
I copied and pasted the codes from notepad again

**Comment by yaaHosein** [comment]
the answer of this question :

This variable meant to hold the value of the last button was clicked
What reason behind setting its value to "+" || "-" || "*" || "/" ?

is 

You're not using _firstOperand in the operator functions
you need to update the value of _firstOperand with the operator type

note
I did not replied the last comment because i am still  now did not finish the above one in this comment

**Comment by SMH110** [comment]
> the answer of this question :
> 
> This variable meant to hold the value of the last button was clicked
> What reason behind setting its value to "+" || "-" || "*" || "/" ?
> 
> is
> 
> You're not using _firstOperand in the operator functions
> you need to update the value of _firstOperand with the operator type
> 
> note
> I did not replied the last comment because i am still now did not finish the above one in this comment

Your operators functions are:
1. [Add](https://github.com/yaaHosein/Calculator1/blob/1074bc1fb0ed9f0ea3fb1bfc81e89ddbd358931b/calculator.js#L105)
2. [Subtract](https://github.com/yaaHosein/Calculator1/blob/1074bc1fb0ed9f0ea3fb1bfc81e89ddbd358931b/calculator.js#L109)
3. [Multiply](https://github.com/yaaHosein/Calculator1/blob/1074bc1fb0ed9f0ea3fb1bfc81e89ddbd358931b/calculator.js#L114)
4. [Divide](https://github.com/yaaHosein/Calculator1/blob/1074bc1fb0ed9f0ea3fb1bfc81e89ddbd358931b/calculator.js#L118)

**Comment by yaaHosein** [comment]
ok Thanks 
sorry for the late reply 
only i had internet to see your comment 
I have pushed some some new updates to to this branch
still now when I click on any number more than one time it is repeated in the console panel only

**Comment by yaaHosein** [comment]
If I set it to as following:

    result.textContent = firstOperandAndOperator;

it will display only the operator in result div

so i added the newValue to it 
so i can see the first operand and the operator

**Comment by SMH110** [comment]
> If I set it to as following:
> 
> ```
> result.textContent = firstOperandAndOperator;
> ```
> 
> it will display only the operator in result div
> 
> so i added the newValue to it
> so i can see the first operand and the operator

If you want to see something you can use 
`console.log()`  and log its value to the console and see it there

can you undo this change as I didn't as for it
Thanks

**Comment by yaaHosein** [comment]
ok thanks 
i did

**Comment by yaaHosein** [comment]
I did not understand this point:

when a user open the app and clicks on 1
_firstOperand should be null or undefined and because _firstOperand is not equal to "+" or "-" or "/"

**Comment by yaaHosein** [comment]
and updated the dividing operator function

File: calculator.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 let lastValue;
 let operator;
+let _Operator;
```
Comment (SMH110) [inline calculator.js]: STOP changing this variable name in every commit.
The best name of this variable is lastButtonClicked

File: calculator.js [hunk 5]
```diff
@@ -98,31 +100,28 @@ negativeValueButton.addEventListener("click", function() {
...
     lastValue = result.textContent;
-    result.textContent = "";
 });
 dividingButton.addEventListener("click", function() {
     operator = "/";
     lastValue = result.textContent;
-    result.textContent = "";
-});
+    // result.textContent = _Operator + "/";
+    result.textContent = _Operator + operator;
```
Comment (SMH110) [inline calculator.js]: Can you remove these two lines

File: calculator.js (hunk 4 above)
Comment (SMH110) [reply calculator.js]: This variable should keep track if the last button was clicked 
for example if a user clicks on button 1 then the value of this variable should be 1 or "NUMBER" or "1"

**Review by SMH110 — COMMENTED** [review]
> I did not understand this point:
> 
> when a user open the app and clicks on 1
> _firstOperand should be null or undefined and because _firstOperand is not equal to "+" or "-" or "/"

When a user opens the app and he didn't click on anything 
what should the value of the `_firstOperand ` be? 
the answer is `null` or `undefined` because this variable meant to hold the value of the last button was clicked and user didn't click on anything yet 
logically the variable value should be `null`

File: calculator.js [hunk 6]
```diff
@@ -104,23 +104,27 @@ negativeValueButton.addEventListener("click", function() {
 
 addingButton.addEventListener("click", function() {
     operator = "+";
-    lastValue = result.textContent;
     lastButtonClicked = "+";
+    lastValue = result.textContent;
+    result.textContent = lastValue + lastButtonClicked;
```
Comment (SMH110) [inline calculator.js]: I don't know why you put this line again `result.textContent = lastValue + lastButtonClicked;` despite a few days ago I told you to remove it.

the operators handlers don't have to modify the result div. they should update some variables.

Comment (yaaHosein) [reply calculator.js]: I putted this time after adding this line :
  lastButtonClicked = "+";

and i putted to use the lastButtonClicked variable in this handler

File: calculator.js (hunk 5 above)
Comment (yaaHosein) [reply calculator.js]: yes I did

File: calculator.js (hunk 4 above)
Comment (yaaHosein) [reply calculator.js]: Ok Thanks

File: calculator.js [hunk 7]
```diff
@@ -106,25 +106,26 @@ addingButton.addEventListener("click", function() {
...
     lastValue = result.textContent;
-    result.textContent = lastValue + lastButtonClicked;
+
 });
 subtractingButton.addEventListener("click", function() {
     operator = "-";
     lastButtonClicked = "-";
     lastValue = result.textContent;
-    result.textContent = lastValue + lastButtonClicked;
+    // lastButtonClicked = lastValue + "-";
```
Comment (SMH110) [inline calculator.js]: Why this commented not removed completely?

File: calculator.js (hunk 6 above)
Comment (SMH110) [reply calculator.js]: you're using the `lastButtonClicked ` in handler

Here you're setting the value of the  `lastButtonClicked ` to "+" ; so you're using it 
`  lastButtonClicked = "+";`

File: calculator.js (hunk 7 above)
Comment (yaaHosein) [reply calculator.js]: As you saw better it is different this time  
I have removed it and pushed the update

File: calculator.js [hunk 8]
```diff
@@ -10,78 +11,79 @@ clearingButton.addEventListener("click", function() {
 });
 
 button0.addEventListener("click", function() {
+    if (lastButtonClicked === "+" || "-" || "*" || "/") result.textContent = "";
```
Comment (SMH110) [inline calculator.js]: This check is wrong.
you need to write this check like this

```js
if (lastButtonClicked === "+"  || lastButtonClicked  === "-" || lastButtonClicked  === "*" || lastButtonClicked  === "/" ) result.textContent = "";
```
and same for button1 - butto9 handlers

**Review by SMH110 — APPROVED** [review]

**Pull request merged**